src/sensor.cpp

```diff
@@ -476,8 +476,20 @@ void log_callback_end( uint32_t fps,
                         }
                         else
                         {
-                            assert( expected_size == sizeof(byte) * f.frame_size );
-                            memcpy( (void *)fh->get_frame_data(), f.pixels, expected_size );
+                            if (req_profile_base->get_format() == RS2_FORMAT_Y12I)
+                            {
+                                assert(expected_size == sizeof(byte) * f.frame_size ||
+                                    expected_size * 3 / 4 == sizeof(byte) * f.frame_size);
+                                if (expected_size == sizeof(byte) * f.frame_size)
+                                    memcpy((void*)fh->get_frame_data(), f.pixels, expected_size);
+                                if (expected_size * 3 / 4 == sizeof(byte) * f.frame_size)
+                                    memcpy((void*)fh->get_frame_data(), f.pixels, expected_size * 3 / 4);
```

Comment:
Maybe we can use a short-handed style:
`expected_size  *= (expected_size * 3 / 4 == sizeof(byte) * f.frame_size)) ?  3/4 : 1;`
and then you can leave the original assert and memcpy in place

Reply:
great idea - thanks